### PR TITLE
PWGHF: Added parametrized pT differential cut in Dplus candidate selector

### PR DESCRIPTION
**Pull request by Luca610**
This PR introduces the possibility to apply a parametrized pT-differential cut in the Dplus candidate selector. This cut riproduces the one present in the trigger implemented in EventFiltering/PWGHF/HFFilterHelpers.h

-In utilsAnalysis.h I implemented a structure with the necessary configurables for this cut and the selection function itself

-In candidateSelectorDplusToPiKPi I added the option to enable this cut

### CODE REVIEW

**Review by mfaggin — APPROVED** [review]

File: PWGHF/Utils/utilsAnalysis.h [hunk 1]
```diff
@@ -108,6 +111,34 @@ bool isSelectedTrackTpcQuality(T const& track, const int tpcNClustersFoundMin, c
...
+  o2::framework::Configurable<o2::framework::LabeledArray<float>> deltaMassPars3Prong{"deltaMassPars3Prong", {defaultDeltaMassPars3Prong[0], 2, {"constant", "linear"}}, "delta mass parameters for HF 3 prong trigger mass cut"};
+  o2::framework::Configurable<o2::framework::LabeledArray<float>> sigmaPars3Prong{"sigmaPars3Prong", {defaultSigmaPars3Prong[0], 2, {"constant", "linear"}}, "sigma parameters for HF 3 prong trigger mass cut"};
+
+  /// Mass selection of D+ candidates to build B0 candidates
+  /// \param pTrackSameChargeFirst is the first same-charge track momentum
+  /// \param pTrackSameChargeFirst is the second same-charge track momentum
+  /// \param pTrackSameChargeFirst is the opposite charge track momentum
+  /// \param ptD is the pt of the D+ meson candidate
+  /// \return true if D+ passes selection
+  bool isSelectedDplusInMassRange(const float& invMassDplus, const float& ptD)
```
Comment (vkucera) [inline PWGHF/Utils/utilsAnalysis.h:131]: What selection does `Selected` represent here? If you only refer to the mass range, it is already expressed by `InMassRange`.

File: PWGHF/Utils/utilsAnalysis.h [hunk 2]
```diff
@@ -108,6 +111,34 @@ bool isSelectedTrackTpcQuality(T const& track, const int tpcNClustersFoundMin, c
...
+  {
+    float peakMean = (ptD < pTMaxDeltaMass) ? ((o2::constants::physics::MassDPlus + deltaMassPars3Prong->get("constant")) + deltaMassPars3Prong->get("linear") * ptD) : o2::constants::physics::MassDPlus;
+    float peakWidth = sigmaPars3Prong->get("constant") + sigmaPars3Prong->get("linear") * ptD;
+
+    if (std::fabs(invMassDplus - peakMean) > nSigma * peakWidth && ptD < pTMaxMassCut) {
+      return false;
+    }
+    return true;
```
Comment (vkucera) [inline PWGHF/Utils/utilsAnalysis.h:139]: This can be a trivial return of the value of the condition itself.

**Review by vkucera — COMMENTED** [review]
@Luca610 What is the need to put this in `PWGHF/Utils/utilsAnalysis.h` instead of keeping it in the selectors? What future development do you plan?

File: PWGHF/Utils/utilsAnalysis.h [hunk 3]
```diff
@@ -108,6 +111,34 @@ bool isSelectedTrackTpcQuality(T const& track, const int tpcNClustersFoundMin, c
...
+  o2::framework::Configurable<float> pTMaxMassCut{"pTMaxMassCut", 8., "Max pT to apply pT differential cut"};
+  o2::framework::Configurable<o2::framework::LabeledArray<float>> deltaMassPars3Prong{"deltaMassPars3Prong", {defaultDeltaMassPars3Prong[0], 2, {"constant", "linear"}}, "delta mass parameters for HF 3 prong trigger mass cut"};
+  o2::framework::Configurable<o2::framework::LabeledArray<float>> sigmaPars3Prong{"sigmaPars3Prong", {defaultSigmaPars3Prong[0], 2, {"constant", "linear"}}, "sigma parameters for HF 3 prong trigger mass cut"};
+
+  /// Mass selection of D+ candidates to build B0 candidates
+  /// \param pTrackSameChargeFirst is the first same-charge track momentum
+  /// \param pTrackSameChargeFirst is the second same-charge track momentum
+  /// \param pTrackSameChargeFirst is the opposite charge track momentum
+  /// \param ptD is the pt of the D+ meson candidate
+  /// \return true if D+ passes selection
```
[the remark this answers is not in the document]
Comment (Luca610) [reply PWGHF/Utils/utilsAnalysis.h:130]: Thank, I missed it

File: PWGHF/Utils/utilsAnalysis.h (hunk 2 above)
Comment (Luca610) [reply PWGHF/Utils/utilsAnalysis.h:139]: You're right, thanks!

**Comment by Luca610** [comment]
Hi @vkucera, Thank you for your comments. I placed the structure in utilsAnalysis because the same cut will be needed also for Ds, Lc and Xc analysis on triggered data.
I will implement your comments in a future PR soon!

**Pull request merged**